File: 121_best_time_to_buy_and_sell_stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProf = 0
        left = 0
        right = 0

        while right < len(prices):
            curProf = prices[right] - prices[left]
            maxProf = max(maxProf, curProf)
            if curProf < 0:
                left += 1
            else:
                right += 1

        return maxProf

File: c++/121_best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int lowest = prices[0];

        for (int n : prices) {
            max_profit = max(max_profit, n - lowest);
            if (n < lowest) {
                lowest = n;
            }
        }

        return max_profit;
    }
};
